nbdime/merging/generic.py

```diff
@@ -383,14 +383,16 @@ def _merge_strings(base, local_diff, remote_diff,
         decisions.conflict(path[:-1],
                            [op_patch(k, local_diff)],
                            [op_patch(k, remote_diff)])
-        _merge_strings.recursion = False
     else:
         # Merge lines as lists
         _merge_strings.recursion = True
         base = base.splitlines(True)
 
-        _merge_lists(
-            base, local_diff, remote_diff, path, decisions)
+        try:
+            _merge_lists(
+                base, local_diff, remote_diff, path, decisions)
+        finally:
+            _merge_strings.recursion = False
```

Comment:
It is not very pretty, but it works (right?). Suggestions for a better pattern are welcome.